### PR TITLE
Firelock ECS + some other stuff.

### PR DESCRIPTION
**Pull request by ElectroJr**
This PR:
- Moves all firelock logic into the system
- Combines IsHoldingPressure and IsHoldingFire into a single function
  - Less iterating over neighbouring tiles & checking for blocked/airtight directions.
- Fixes pressure & fire checks. 
  - Previously these just didn't work because a fire lock would only iterate over un-blocked adjacent tiles, but the airlock itself ensured that there were no unblocked tiles.
  - The way it does these checks now might be somewhat questionable. This will probably also needs rejigging for the planet/atmosphere stuff. @Zumorica
- Makes the FirelockSystem periodically check if a firelock is holding back pressure or a fire, and will turn on lights if it is.
  - The pressure check is only done for closed firelocks, but does still involve periodically querying & iterating over all firelock entities
  - Might be better to give firelocks an `ActiveFirelockVisuals` component whenever they get closed, and iterate over that instead?
- Some sprites were changed
  - Some blank `closed_unlit` states were fixed.
  - Shuttle airlock `closed_unlit` and `bolted_unlit` were swapped, to make them consistent with other airlocks (red for bolted, orange/yellow for closed).
  - There is now an appearance data enum to toggle showing the `closed_unlit` state.

The actual reason this PR exists is because I wanted to start working on proper pressure cycling airlocks, and was working on a pressure-sensing component for doors that would turn a light on and block opening if there was a significant pressure difference, but then I realized that firelocks basically already did all this and you could just slap a firelock onto an external airlock. But then again, maybe some of the firelock functionality should just be split off into different components.

Demo video showing the is-holding-pressure checks in action. I haven't actually tested fire. Video also shows some external airlocks that were given a firelock component, but the entity prototypes weren't actually changed.

https://user-images.githubusercontent.com/60421075/169737147-f32ae871-b247-4814-9a8a-46cc0e225710.mp4

Note that the video shows the wrong pop-ups (fire warning instead of pressure warning). This has since been fixed.

:cl:
- fix: Firelocks will now properly warn users that try to open them if they are holding back an inferno or the cold vacuum of space.

### CODE REVIEW

File: Content.Server/Doors/Systems/FirelockSystem.cs [hunk 1]
```diff
@@ -1,77 +1,159 @@
...
+            var powerQuery = GetEntityQuery<ApcPowerReceiverComponent>();
+            var airtightQuery = GetEntityQuery<AirtightComponent>();
+
+            foreach (var (firelock, door, appearance, xform) in EntityQuery<FirelockComponent, DoorComponent, AppearanceComponent, TransformComponent>())
+            {
+                if (!airtightQuery.TryGetComponent(door.Owner, out var airtight))
+                    continue;
+
+                UpdateVisuals(door.Owner, firelock, door, airtight, appearance, xform, powerQuery, airtightQuery);
+            }
```
Comment (metalgearsloth) [inline Content.Server/Doors/Systems/FirelockSystem.cs:65]: This looks expensive?

**Comment by wrexbe** [comment]
I think the firelock tried to close on me, and became both open, and airtight at the same time.

airtight

File: Content.Server/Doors/Systems/FirelockSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/Doors/Systems/FirelockSystem.cs]: Does it need to update regularly because of atmos?

**Review by mirrorcult — APPROVED** [review]
needs conflicts addressed

**Comment by wrexbe** [comment]
- fixed merge conflict
- had to replace CheckPressureAndFire with not as optimized version, was no longer compatible with master.
- fixed bug with EmergencyPressureStop making the door open, but air tight.

Played with it for a bit, and it seems to work fine now. Going to merge it like this, instead of it being stuck for eternity.

**Pull request merged**